flow/record/base.py

```diff
@@ -661,7 +679,55 @@ def DynamicDescriptor(name, fields):
     return RecordDescriptor(name, [("dynamic", field) for field in fields])
 
 
-def open_path(path, mode, clobber=True):
+def open_stream(fp, mode) -> IO | Peekable:
```

Comment:
Maybe you can subclass `Peekable` from an IO class, then this type hint can be more generic / just IO..